src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1470,9 +1470,12 @@ private void ReportNullabilityMismatchWithTargetDelegate(SyntaxNode syntax, Name
                                 return true;
                             }
                         }
+                        else if (sourceTypeOpt?.IsReferenceType == true)
+                        {
+                            return true;
```

Comment:
>return true; [](start = 28, length = 12)

Should we simply return ```operandIsNotNull```? It feels like the last adjustment wasn't accompanied by tests. Can we test this code path?